osgp/protocol-adapter-iec60870/osgp-protocol-simulator-iec60870/src/main/java/org/opensmartgridplatform/simulator/protocol/iec60870/domain/profile/DefaultControlledStationAsduFactory.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Configuration
+@Profile("default_controlled_station")
+public class DefaultControlledStationAsduFactory implements Iec60870AsduFactory {
+
+    @Value("${general_interrogation_object_addresses}")
+    private int[] ioa;
+
+    @Value("${general_interrogation_element_values}")
+    private float[] iev;
+
+    @Override
+    public ASdu createInterrogationCommandResponseAsdu(final long timestamp) {
+        final InformationObject[] informationObjects = new InformationObject[this.ioa.length];
+        for (int index = 0; index < this.ioa.length; index++) {
+            informationObjects[index] = new InformationObject(this.ioa[index],
+                    this.createInformationElement(this.iev[index], timestamp));
+        }
+
+        return new Iec60870ASduBuilder().withTypeId(ASduType.M_ME_TF_1)
+                .withSequenceOfElements(false)
+                .withCauseOfTransmission(CauseOfTransmission.SPONTANEOUS)
+                .withInformationObjects(informationObjects)
+                .build();
+    }
+
+    private InformationElement[][] createInformationElement(final float value, final long timestamp) {
+        return new InformationElement[][] { { new IeShortFloat(value), new IeQuality(false, false, false, false, false),
+                new IeTime56(timestamp) } };
+    }
+
+    public ASdu createSingleCommandAsdu() {
+        return new Iec60870ASduBuilder().withTypeId(ASduType.C_SC_NA_1)
+                .withCauseOfTransmission(CauseOfTransmission.SPONTANEOUS)
+                .withInformationObjects(new InformationObject[] { new InformationObject(0,
+                        new InformationElement[][] { { this.defaultIeQualifierOfInterrogation() } }) })
+                .build();
+    }
+
+    public ASdu createShortFloatingPointMeasurementAsdu() {
+        final ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
+        final long timestamp = now.toInstant().toEpochMilli();
+        final float hour = now.getHour();
+        final float minute = now.getMinute();
+
+        final InformationObject[] informationObjects = new InformationObject[this.ioa.length];
+        for (int index = 0; index < this.ioa.length; index++) {
+            final float value = (index == 0 ? hour : minute);
+            informationObjects[index] = new InformationObject(this.ioa[index],
+                    this.createInformationElement(value, timestamp));
+        }
+
+        return new Iec60870ASduBuilder().withTypeId(ASduType.M_ME_TF_1)
+                .withSequenceOfElements(false)
+                .withCauseOfTransmission(CauseOfTransmission.SPONTANEOUS)
+                .withInformationObjects(informationObjects)
+                .build();
+
+    }
+
+    // TODO: remove these setters when the cucumber mock server can read these
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1135)